python-avd/pyavd/schema/generate_docs/tablerowgen.py

```diff
@@ -0,0 +1,355 @@
+        elif self.schema.deprecation.remove_after_date:
+            descriptions.append(f"Support {removed_verb} removed in the first major AVD version released after {self.schema.deprecation.remove_after_date}.")
+        elif self.schema.deprecation.removed:
+            descriptions.append("Support was removed in AVD.")
+
+        if self.schema.deprecation.new_key:
+            descriptions.append(f"Use <samp>{self.schema.deprecation.new_key}</samp> instead.")
+
+        if self.schema.deprecation.url:
+            descriptions.append(f"See [here]({self.schema.deprecation.url}) for details.")
+
+        description = " ".join(descriptions)
+        return f'<span style="color:red">{description}</span>'
+
+    def render_key(self) -> str:
+        """
+        Renders markdown for "key" field including mouse-over and deprecation label with color.
+        """
+        path = ".".join(self.schema._path)
+
+        if self.schema._key:
+            key = self.schema._key.replace("<", "&lt;").replace(">", "&gt;")
+        else:
+            key = f"&lt;{self.schema.type}&gt;"
+
+            # TODO: Remove legacy output
+            if LEGACY_OUTPUT:
+                path += f".{key}"
+
+        if LEGACY_OUTPUT:
+            path = path.replace("<", "&lt;").replace(">", "&gt;")
+
+        return f'[<samp>{self.get_indentation()}{key}</samp>](## "{path}"){self.get_deprecation_label()}'
+
+    def render_type(self) -> str:
+        """
+        Renders markdown for "type" field.
+        """
+        type_converters = {
+            "str": "String",
+            "int": "Integer",
+            "bool": "Boolean",
+            "dict": "Dictionary",
+            "list": "List",
+        }
+        return type_converters[self.schema.type]
+
+    def render_required(self) -> str | None:
+        """
+        Render markdown for "required" field.
+        """
+        if self.schema._is_primary_key:
+            return "Required, Unique"
+        if self.schema.required:
+            return "Required"
+
+    def render_default(self) -> str | None:
+        """
+        Should render markdown for "default" field.
+        """
+        if self.schema.default is not None:
+            if isinstance(self.schema.default, (list, dict)) and len(self.schema.default) > 1:
+                return "See (+) on YAML tab"
+
+            return f"`{self.schema.default}`"
+
+    def render_description(self) -> str | None:
+        """
+        Renders markdown for "description" field including deprecation text with color.
+        """
+        descriptions = []
+        if self.schema.description:
+            descriptions.append(self.schema.description.replace("\n", "<br>"))
+        description = " ".join(descriptions)
+
+        if deprecation := self.get_deprecation_description():
+            description += deprecation
+
+        return description or None
```

Comment:
Somehow it would make more sense for me to write it like this (get the full list in descriptions and then join everything)

```suggestion
        if deprecation := self.get_deprecation_description():
            descriptions.append(deprecation)
       
        return " ".join(descriptions) or None
```

Reply:
Did something like this (which was also the original plan. Issue was that the current md files did not have a space between the description and the deprecation, so I changed it later.)